src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -0,0 +1,302 @@
+            'unity' => $product->unity,
+            'additional_shipping_cost' => $product->additional_shipping_cost,
+        ];
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return array<string, int|bool|null>
+     */
+    private function getStockFields(Product $product): array
+    {
+        return [
+            'minimal_quantity' => (int) $product->minimal_quantity,
+            'low_stock_threshold' => $product->low_stock_threshold,
+            'low_stock_alert' => (int) $product->low_stock_alert,
+            'advanced_stock_management' => $product->advanced_stock_management,
+            'pack_stock_type' => (int) $product->pack_stock_type,
+        ];
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return array<string, int|string>
+     */
+    private function getSEOFields(Product $product): array
+    {
+        return [
+            'redirect_type' => $product->redirect_type,
+            'id_type_redirected' => (int) $product->id_type_redirected,
+        ];
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return array<string, int|string|bool>
+     */
+    private function getOptionsFields(Product $product): array
+    {
+        return [
+            'condition' => $product->condition,
+            'show_condition' => (int) $product->show_condition,
+            'show_price' => (int) $product->show_price,
+            'visibility' => $product->visibility,
+            'on_sale' => (int) $product->on_sale,
+            'online_only' => (int) $product->online_only,
+            'active' => (int) $product->active,
+            'available_for_order' => (int) $product->available_for_order,
+            'indexed' => (int) $product->indexed,
+        ];
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return array<string, int>
+     */
+    private function getCustomizationFields(Product $product): array
+    {
+        return [
+            'customizable' => (int) $product->customizable,
+            'uploadable_files' => (int) $product->uploadable_files,
+            'text_fields' => (int) $product->text_fields,
+        ];
+    }
+
+    /**
+     * Date fields are all mandatory.
+     *
+     * @param Product $product
+     *
+     * @return array<string, string|null>
+     */
+    private function getDateFields(Product $product): array
+    {
+        $now = new DateTime();
+
+        return [
+            'date_add' => $product->date_add,
+            'date_upd' => $now->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT),
+            'available_date' => DateTimeUtil::getNullableDateTime($product->available_date),
+        ];
+    }
+
+    /**
+     * @param array $fields
+     *
+     * @return array
+     */
+    private function formatQueryValues(array $fields): array
+    {
+        $values = [];
+        foreach (array_keys($fields) as $column) {
+            $values[sprintf('`%s`', $column)] = sprintf(':%s', $column);
```

Comment:
```suggestion
            $values[sprintf('`%s`', bqSQL($column))] = sprintf(':%s', $column);
```
?

Reply:
In this case I'm not sure it's required because this time it is based on a Doctrine query